arch/arm/core/aarch64/smp.c

```diff
@@ -0,0 +1,129 @@
+/*
+ * Copyright 2020 NXP
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ */
+
+/**
+ * @file
+ * @brief codes required for AArch64 multicore and Zephyr smp support
+ */
+
+#include <cache.h>
+#include <device.h>
+#include <kernel.h>
+#include <kernel_structs.h>
+#include <ksched.h>
+#include <soc.h>
+#include <init.h>
+#include <arch/arm/aarch64/arm_mmu.h>
+#include <drivers/interrupt_controller/gic.h>
+#include <drivers/psci.h>
+#include <sys/arch_interface.h>
+
+volatile struct {
+	arch_cpustart_t fn;
+	void *arg;
+} arm64_cpu_init[CONFIG_MP_NUM_CPUS];
+
+volatile char *arm64_cpu_sp;
```

Comment:
I'm a bit lost on this. Shouldn't this be one for each CPU?

Reply:
No. secondary cores are started one by one, so this variable could be reused.